Data/IntMap/Internal.hs

```diff
@@ -980,6 +984,22 @@ alterF f k m = (<$> f mv) $ \fres ->
     Nothing -> maybe m (const (delete k m)) mv
     Just v' -> insert k v' m
   where mv = lookup k m
+-- TODO(m-renaud): Figure out if this should be marked INLINE or NOINLINE.
+-- It needs to be one or the other or else the specialization rule may not fire.
+{-# NOINLINE [1] alterF #-}
+#if MIN_VERSION_base(4,8,0)
+{-# RULES
+"Identity specialize alterF" forall (f :: Maybe a -> Identity (Maybe a)) k m.
+  alterF f k m = Identity $ alter (runIdentity . f) k m
+  #-}
+#endif
+
+#if MIN_VERSION_base(4,9,0)
+{-# RULES
+"Const specialize alterF" forall (f :: Maybe a -> Const x (Maybe a)) k m.
+  alterF f k m = Const . getConst . f $ lookup k m
```

Comment:
Is the surprising `Const . getConst` needed to change a phantom?

Reply:
Used my new-found knowledge of `coerce` :)